heron/packing/tests/java/com/twitter/heron/packing/roundrobin/ResourceCompliantRRPackingTest.java

```diff
@@ -0,0 +1,570 @@
+
+
+    topologyConfig.setContainerRamRequested(containerRam);
+
+
+    TopologyAPI.Topology topologyExplicitRamMap =
+        getTopology(spoutParallelism, boltParallelism, topologyConfig);
+    PackingPlan packingPlan =
+        getResourceCompliantRRPackingPlan(topologyExplicitRamMap);
+    Assert.assertEquals(packingPlan, null);
+  }
+
+  /**
+   * Test the scenario where the user defined number of containers is not sufficient and
+   * automatic adjustment has been enabled.
+   */
+  @Test
+  public void testInsufficientContainers2() throws Exception {
+    int numContainers = 1;
+    int spoutParallelism = 4;
+    int boltParallelism = 3;
+
+    // Set up the topology and its config
+    com.twitter.heron.api.Config topologyConfig = new com.twitter.heron.api.Config();
+    topologyConfig.put(com.twitter.heron.api.Config.TOPOLOGY_STMGRS, numContainers);
+
+    // Explicit set resources for container
+    long containerRam = 2L * Constants.GB;
+
+    topologyConfig.setContainerRamRequested(containerRam);
+    topologyConfig.setAdjustNumberOfContainers(true);
+
+    TopologyAPI.Topology topologyExplicitRamMap =
+        getTopology(spoutParallelism, boltParallelism, topologyConfig);
+    PackingPlan packingPlan =
+        getResourceCompliantRRPackingPlan(topologyExplicitRamMap);
+    Assert.assertEquals(packingPlan.containers.size(), 4);
+
+    Assert.assertEquals((long) (Math.round(2 * instanceCpuDefault
+            + 10 * (2 * instanceCpuDefault) / 100)
+            + Math.round(2 * instanceCpuDefault
+            + 10 * (2 * instanceCpuDefault) / 100)
+            + Math.round(2 * instanceCpuDefault
+            + 10 * (2 * instanceCpuDefault) / 100)
+            + Math.round(1 * instanceCpuDefault
+            + 10 * (1 * instanceCpuDefault) / 100)
+            + instanceCpuDefault),
+        (long) packingPlan.resource.cpu);
+
+    Assert.assertEquals(2 * instanceRamDefault
+            + 10 * (2 * instanceRamDefault) / 100
+            + 2 * instanceRamDefault
+            + 10 * (2 * instanceRamDefault) / 100
+            + 2 * instanceRamDefault
+            + 10 * (2 * instanceRamDefault) / 100
+            + 1 * instanceRamDefault
+            + 10 * (1 * instanceRamDefault) / 100
+            + instanceRamDefault,
+        packingPlan.resource.ram);
+
+    Assert.assertEquals(2 * instanceDiskDefault
+            + 10 * (2 * instanceDiskDefault) / 100
+            + 2 * instanceDiskDefault
+            + 10 * (2 * instanceDiskDefault) / 100
+            + 2 * instanceDiskDefault
+            + 10 * (2 * instanceDiskDefault) / 100
+            + 1 * instanceDiskDefault
+            + 10 * (1 * instanceDiskDefault) / 100
+            + instanceDiskDefault,
+        packingPlan.resource.disk);
+  }
+
+  /**
+   * Test the scenario where the user defined number of containers is not sufficient and
+   * automatic adjustment has been enabled.
+   */
+  @Test
+  public void testInsufficientContainers3() throws Exception {
```

Comment:
could you either name testInsufficientContainers[1|2|3] more descriptively or clarify in their javadocs what makes them different.